cluster/gce/gci/configure-helper.sh

```diff
@@ -242,16 +244,19 @@ function safe-format-and-mount() {
 # Gets a devices UUID and bind mounts the device to mount location in
 # /mnt/disks/by-id/
 function unique-uuid-bind-mount(){
-  local mountpoint="${1}"
-  local actual_device="${2}"
+  local mountpoint
+  local actual_device
+  mountpoint="$1"
+  actual_device="$2"
 
   # Trigger udev refresh so that newly formatted devices are propagated in by-uuid
   udevadm control --reload-rules
   udevadm trigger
   udevadm settle
 
   # grep the exact match of actual device, prevents substring matching
-  local myuuid=$(ls -l /dev/disk/by-uuid/ | grep "/${actual_device}$" | tr -s ' ' | cut -d ' ' -f 9)
+  local myuuid
+  myuuid=$(find -L /dev/disk/by-uuid -maxdepth 1 -samefile /dev/"${actual_device}" -printf "%P\n")
```

Comment:
can you explain why the change here? I'm not confident this is the same as before

Reply:
This patch is about taking care of the shellcheck warnings and errors of the current code. (Splitting the local part to a separate line goes without saying.)
The current ls | grep thing triggers [SC2010](https://github.com/koalaman/shellcheck/wiki/SC2010) and needs to be changed in order not to.
I figured instead of using these 4 different programs and 3 pipes, it would be more convenient to use 1 single well-known standard program (find) instead.

The current code does `ls -l` on `/dev/disk/by-uuid` which gives a result similar to

```
lrwxrwxrwx 1 root root 10 maj 16 08:18 0493e0d4-53c8-4f46-a339-f9e147a6e321 -> ../../dm-5
lrwxrwxrwx 1 root root 10 maj 16 08:18 1dcd0d80-2b96-4b1a-8890-881d20d59af6 -> ../../dm-6

```
[etc.]

and the greps `${actual_device} `to filter out the device we are searching the uuid for, pipes it to tr to replace all multiple occurrences of space with a single one and then pipes it to cut to use space as the delimiter and putput the 9th column, which is the uuid. For the `dm-5` above, this would be `0493e0d4-53c8-4f46-a339-f9e147a6e321`.

Now what the find command here does is to instead to search `/dev/disk/by-uuid` (with depth 1) and match the symlink with the `/dev/${actual_device}` that we are looking for (in this example above, match the symlink in `/dev/disk/by-uuid` with the `/dev/dm-5` which is `0493e0d4-53c8-4f46-a339-f9e147a6e321`, and then print the result with find's %P print format (file's name with the name of the starting-point under which it was found removed).

If you try both lines of code, you should see that they produce the same output. And, of course, the find solution passes shellcheck.

I would suggest that this is more robust than relying on the ls output.